### ISSUE TITLE
🚀 Feature: Properly finish scheduled tasks if the app gets SIGTERM signal

### ISSUE THREAD

**Issue by bk201-** [backstage/backstage#23274]
### 🔖 Feature description

If the Backstage app receives the SIGTERM/SIGINT signal all scheduled and running tasks should be properly finished, to start again after restarting the app. 

### 🎤 Context

When the application is killed, for example, the pod terminates with the backstage app, all running scheduled tasks don't handle this, therefor the field "current_run_ticket" in the DB won't be set to NULL. In this case, the next run will never happen it was described here #11526. 

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by freben** [comment]
Waiting until all tasks are finished is tricky to do generally - some people schedule pretty long running things and you have to bear in mind that this is a blocker for rollouts. It's unfortunate if you're in an incident scenario and your service kinda refuses to go down in a timely fashion for the rollout of the patched version.

Task functions receive an `AbortSignal` as their first argument. They are meant to shut themselves down ASAP when that signal triggers. This happens at task timeout for example. However, I don't believe that we've attached the lifecycle service to triggering that. There's probably more that can be done to ensure that all running tasks are instantly informed of an impending shutdown.

We also don't have a heartbeat mechanism built in yet, so if a task worker dies entirely, the timeout period must pass before the next attempt is made.

The tricky thing with "forcing" timeouts when we don't know if the work is actually ongoing still or not, is that we pretty strongly try to avoid any overlapping work. Because a lot of the work is of such a nature that it updates database tables and don't handle overlaps well.

In summary:

- Actually blocking for long times will probably not be the right solution
- Being better about aggressively triggering the abort signal on shutdown (based on the lifecycle service) would be a worthwhile improvement PR
- Introducing a heartbeat mechanism for workers that let us set the ticket to null faster than the timeout period would be a worthwhile improvement PR

Adding a help wanted label on this

**Comment by freben** [comment]
The ticket [does get set to null](https://github.com/backstage/backstage/blob/bf2c435983b18cc6674c226b82d2694557e53252/packages/backend-tasks/src/tasks/PluginTaskSchedulerJanitor.ts#L69) by the janitor at timeout expiration.

My suggestion about heartbeats means that we'll be able to do the same _faster_ than the timeout expiration, if the task is badly behaved and doesn't respect the abort signal well.
